### ISSUE TITLE
FIM monitoring of an entire drive in real time 

### ISSUE THREAD

**Issue by ImGyba** [wazuh/wazuh#21745]
Hello,

I am using Wazuh 4.7.2 and I want to monitor the entire C:\ drive on a Windows endpoint.

In **ossec.conf**, on the Windows endpoint, after looking over a few similar issues found here on GitHub, I tried multiple ways to set the path:

`<directories realtime="yes" check_all="yes" whodata="yes" report_changes="yes">C:\*</directories>`
`<directories realtime="yes" check_all="yes" whodata="yes" report_changes="yes">C:\.</directories>`
`<directories realtime="yes" check_all="yes" whodata="yes" report_changes="yes">C:\\</directories>`
`<directories realtime="yes" check_all="yes" whodata="yes" report_changes="yes">C:\</directories>`

None of them works.   

Reading through other people's issues, I saw that there is a parameter in the **internal_options.conf** file that limits the number of folders that can be monitored in real-time. 

```
# Maximum number of directories monitored for realtime on windows [1..1024]
syscheck.max_fd_win_rt=256
```

Is this why the real-time FIM monitoring is not working for an entire drive? Or can you set this up in another way? 

Any suggestion is welcomed.

Thank you.

**Comment by vikman90** [comment]
@ImGyba To configure the agent for monitoring your entire drive, use the following command:

```xml
<directories check_all="yes" whodata="yes" report_changes="yes">C:</directories>
```

The option `whodata=yes` takes precedence over `realtime=yes`. When who-data is enabled, the agent sets up auditing configurations for the monitored folders. In contrast, the real-time mode establishes a recursive watch in the directory tree rooted by the specified drive (e.g., C:). Consequently, there is no need to increase the value of `syscheck.max_fd_win_rt.`

## Considerations for monitoring an entire drive

- Monitoring an entire drive may lead to tracking a substantial number of files, potentially impacting system resources.
- You might need to extend or disable the file limit (`<file_limit>`) to accommodate the increased monitoring scope.
- Adjust the `<max_files_per_second>` setting to ensure efficient scanning without undue delays.
- If you enable the `report_changes=yes` option, be aware that it creates backups of each monitored file, potentially duplicating disk usage. Before enabling this option, verify that your disk usage remains below 50%.

### Recursive bombs

Exercise caution when monitoring critical directories like the Windows or agent folder:

- Monitoring the agent’s log may trigger repeated alerts and logs whenever the log changes.
- Monitoring the agent’s `diff` folder could lead to recursive backups of backups, potentially filling up your disk.
- Monitoring the Windows folder (which contains audit logs) may generate additional audit logs when changes occur.

In summary, while monitoring an entire drive is feasible, I recommend focusing on specific directories rather than setting the agent to watch the entire C: drive in real-time or who-data mode with change reporting enabled.

Best regards.

**Comment by sontinh2021** [comment]
<directories check_all="yes" whodata="yes" report_changes="yes">C:</directories>
Can not run on window wazuh agent version 4.7.3

**Comment by Esther7171** [comment]
can i use same for virus total  ??
